[.NET Profiling] Better explain cpu and memory overhead (#18942)
DataDog/documentation: content/en/profiler/profiler_troubleshooting/dotnet.md

## Reduce overhead when using the profiler

> The different profilers have a fixed cpu and memory overhead. Because such a fixed overhead per profiled application exists, *the more profiled applications, the higher the overhead!*

Reviewer: `The different profilers` may be confusing to customers, they could understand `the different runtime profilers`

We could reword into `The different profile types` with a link to .NET profile types list

### Avoid enabling the profiler machine-wide

Datadog does not recommend enabling the profiler at machine-level or for all IIS application pools. In order to reduce the amount of resources used by the profiler, you can:
- Increase the allocated resources, such as increasing CPU cores.
- Profile only specific applications by setting environment in batch files instead of directly running the application.
- Reduce the number of IIS pools being profiled (only possible in IIS 10 or later).
- Disable wall time profiling with the setting `DD_PROFILING_WALLTIME_ENABLED=0`.

> ### Linux Containers

Reviewer 2: More as a question to @gturbat, should we move this to be more generic to profiling? E.g. I guess a few of our profilers can have a bigger impact on really really tiny containers, not just .net?

Reviewer: I assume you meant to comment the header  `### Linux Containers` below?

## Reduce overhead when using the profiler

> The [different profile types][3] have a fixed CPU and memory overhead. Because such a fixed overhead per profiled application exists, *the more profiled applications, the higher the overhead.*

Reviewer 3: ```suggestion
The [different profile types][3] have a fixed CPU and memory overhead, so *the more profiled applications, the higher the overhead.*
```

### Avoid enabling the profiler machine-wide

> Datadog does not recommend enabling the profiler at machine-level or for all IIS application pools. In order to reduce the amount of resources used by the profiler, you can:

Reviewer 3: ```suggestion
Datadog does not recommend enabling the profiler at machine-level or for all IIS application pools. To reduce the amount of resources used by the profiler, you can:
```

- Increase the allocated resources, such as increasing CPU cores.
- Profile only specific applications by setting environment in batch files instead of directly running the application.
- Reduce the number of IIS pools being profiled (only possible in IIS 10 or later).
- Disable wall time profiling with the setting `DD_PROFILING_WALLTIME_ENABLED=0`.

### Linux Containers

The exact value can vary but the fixed overhead cost means that the relative overhead of the profiler can be significant in very small containers. To avoid this situation, the profiler is disabled in containers with less than one core. You can override the one core threshold by setting the `DD_PROFILING_MIN_CORES_THRESHOLD` environment variable to a value less than one. For example, a value of `0.5` allows the profiler to run in a container with at least 0.5 cores.

> However, in that case, you should expect a CPU consumption increase; even for idle services because the profiler threads will always scan the application's threads. The less available core, the more the CPU consumption will increase. 

Reviewer 3: ```suggestion
However, in that case, there will be a CPU consumption increase, even for idle services, because the profiler threads always scan the application's threads. The less available core, the more the CPU consumption increases. 
```

> Disabling the wall time profiler with the setting `DD_PROFILING_WALLTIME_ENABLED=0` will decrease the CPU consumption due to the profiler. If this is not enough, increase the CPU cores available for your containers.

Reviewer 3: ```suggestion
Disabling the wall time profiler with the setting `DD_PROFILING_WALLTIME_ENABLED=0` decreases the CPU consumption by the profiler. If this is not enough, increase the CPU cores available for your containers.
```
